Lib/urllib/request.py

```diff
@@ -881,15 +881,15 @@ def reduce_uri(self, uri, default_port=True):
         return authority, path
 
     def is_suburi(self, base, test):
-        """Check if test is below base in a URI tree
+        """Check if test is equal or below base in a URI tree
 
         Both args must be URIs in reduced form.
         """
         if base == test:
             return True
-        if base[0] != test[0]:
+        if base[0] != test[0] or len(test[1]) < len(base[1]):
```

Comment:
The test cases pass even if this change is not included. Are you sure it's needed?